polaris-migrator/src/migrations/v12-react-replace-link-components/v12-react-replace-link-components.ts

```diff
@@ -0,0 +1,79 @@
+import type {API, FileInfo, JSXAttribute, JSXOpeningElement} from 'jscodeshift';
+
+import {
+  insertJSXAttribute,
+  insertJSXComment,
+  removeJSXAttributes,
+  renameProps,
+} from '../../utilities/jsx';
+import {POLARIS_MIGRATOR_COMMENT} from '../../constants';
+
+export default function v12ReactReplaceLinkComponents(
+  fileInfo: FileInfo,
+  {jscodeshift: j}: API,
+) {
+  const source = j(fileInfo.source);
+
+  const localElementName = 'Link';
+
+  source.findJSXElements(localElementName).forEach((element) => {
+    const allAttributes =
+      (j(element).find(j.JSXOpeningElement).get().value as JSXOpeningElement)
+        .attributes ?? [];
+
+    if (allAttributes.some((attribute) => attribute.type !== 'JSXAttribute')) {
+      insertJSXComment(j, element, POLARIS_MIGRATOR_COMMENT);
+      return;
+    }
+
+    const jsxAttributes = allAttributes as JSXAttribute[];
+
+    const monochromeAttribute = jsxAttributes.find(
+      (attribute) => attribute.name.name === 'monochrome',
+    );
+
+    if (monochromeAttribute) {
+      if (!monochromeAttribute || monochromeAttribute.value !== null) {
```

Comment:
Can this be reduced to:

```suggestion
    if (monochromeAttribute && monochromeAttribute.value !== null) {
```

e.g. 
```js
// If `monochrome` is present but NOT an empty boolean prop, exit early
if (monochromeAttribute && monochromeAttribute.value !== null) {
  // <Link monochrome={false} />
  // <Link monochrome={getMonochrome()} />
  // <Link monochrome={condition ? true : false} /> // etc.
  return
}
// <Link /> or <Link monochrome />

// Past this point, we are in the "happy path" where `monochrome` can
// be evaluated from it's presence
const isMonochrome = !!monochromeAttribute
```

Reply:
Good call! ✨